src/docs/sdk/performance/ui-event-transactions.mdx

```diff
@@ -0,0 +1,173 @@
+scroll events, pinches, etc. The UI events the SDK can hook into might vary depending on the
+platform. UI event transactions are an expansion to transactions generated automatically by
+the SDK, which are referred to as auto-generated transactions in this document. 
+
+Creating transactions for a single UI event without any spans would be fruitless. Instead, the
+SDK shall use a UI event as the entry point to a possibly meaningful transaction. It should
+wait to see if it can add any auto-generated spans to the transaction. If it can, it shall keep
+and send the transaction. If not, the SDK should discard the empty transaction. A combination of
+two concepts is needed to implement the wait-and-see logic: __idle transactions__ and
+__wait-for-children__. Check out the specification below to see how those two concepts work
+together with covering multiple edge cases. The following description is drastically simplified:
+
+* __Idle-transactions__: The SDK starts an idle timeout for the transaction when starting it.
+When the transaction or any of its spans starts a new span, it resets the timeout. The SDK finishes
+the transaction when the timeout succeeds.
+* __Wait-for-children__: A transaction waits for all its child spans to finish before finishing itself.
+
+Before diving into the specification with all the edge cases, let's look at two simple examples:
+
+1. The user clicks a button that triggers some requests to the backend and stores data in the
+local database. The SDK can create a meaningful transaction in that case.
+2. The user clicks a button that solely validates some form data and triggers nothing that the
+SDK can automatically instrument. The SDK could only create a transaction without any spans, which would be valueless.
+
+# Specification
+
+Users can change the `idleTimeout` via the SDK config options. The default is 3.0 seconds. 
+
+The specification is written in the [Gherkin syntax](https://cucumber.io/docs/gherkin/reference/).
+
+**1. Starting UI Event transactions** 
+```
+Given an instrumentable UI event
+Then the SDK starts a UI event transaction 
+And starts the idle timeout with idleTimeout of the options
+```
+
+**5. Wait for children when starting span**
+```
+Given a UI event transaction
+When the SDK starts a child span
+Then the SDK cancels the idle timeout
+And waits for the child to finish
+```
+
+**4. Start timeout when the last span finishes**
+```
+Given a UI event transaction
+And the transaction has one or multiple running child spans
+And the transaction is waiting for its children to finish
+When the SDK finishes the last child span
+Then the SDK starts the idle timeout
+```
+
+**3. Don't reset timeout when the second last span finishes**
+```
+Given a UI event transaction
+And the transaction has two running child spans
+When the SDK finishes the first child span
+Then the SDK doesn't reset the idle timeout
+And waits for the children to finish
```

Comment:
H: This is different than Android and Flutter, actually, when a new span is created, the timer is reset only.
@romtsn to confirm.

Reply:
Doesn't Android also use `waitForChildren` @romtsn ?

Reply:
There's `waitForChildren`, but the approach here is different if I understood it correctly, it does reset the timer every time that a new child is created and so on, at the end, the result is the same, I guess.

Reply:
I guess I'm confused a bit by the word `reset` - does it mean `cancel` or does it mean `reschedule`? If we mean `reschedule` here, I think we can remove this scenario, because it should be covered by the one above ("when the _last_ span finishes")

That's what we do on Android (if all children are finished - we reschedule the finish timer):
https://github.com/getsentry/sentry-java/blob/84f0ef1abb55bccd90315665ff4e307c5f71886c/sentry/src/main/java/io/sentry/SentryTracer.java#L234-L245

Reply:
It's `reschedule`. 

>  I think we can remove this scenario, because it should be covered by the one above

Yes, it's covered. I removed it.